prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -87,17 +64,36 @@ fun Observation.addMappedConditions(metadata: Metadata): List<ActionLogDetail> {
     val codeSourcesMap = this.getCodeSourcesMap().filterValues { it.isNotEmpty() }
     var mappedSomething = false
     if (codeSourcesMap.values.flatten().isEmpty()) return listOf(UnmappableConditionMessage()) // no codes found
-
+    val mappingTable = metadata.findLookupTable("observation-mapping").also {
+        if (it == null) { // could not load the table
+            throw IllegalStateException("Unable to load lookup table 'observation-mapping' for condition stamping")
+        }
+    }!!
+    val codes = codeSourcesMap.flatMap { it.value.map { code -> code.code } }
```

Comment:
```suggestion
    val codes = codeSourcesMap.values.flatten().mapNotNull { it.code }
```

small thing but figured it was a little better to use `flatten()` since it is used in the code on line 66